### PR TITLE
Make ModuleArgsParser more understandable

### PR DESCRIPTION
**Pull request by willthames**
Both comments and method names for handling new/old
style parameters were switched around

Made comments and method names reflect actual code paths
taken.

### CODE REVIEW

File: lib/ansible/modules/extras [hunk 1]
```diff
@@ -1 +1 @@
-Subproject commit f798240f436a16a828f48759bbd176b6bccdfe75
+Subproject commit b20cb4dd71425a48e68e0af06039afd2a410cb13
```
Comment (bcoca) [inline lib/ansible/modules/extras]: someone snuck in submodule ref changes ?

**Comment by willthames** [comment]
@bcoca, oops, I updated my commit and it pulled those in. I've removed them now. 'snuck in' is probably an ungenerous term.

**Comment by bcoca** [comment]
sorry, i get cranky after a full day of triage

**Comment by willthames** [comment]
ready_for_review

File: lib/ansible/parsing/mod_args.py [hunk 2]
```diff
@@ -178,15 +178,15 @@ def _normalize_parameters(self, thing, action=None, additional_args=dict()):
 
         return (action, final_args)
 
-    def _normalize_old_style_args(self, thing, action):
+    def _normalize_new_style_args(self, thing, action):
         '''
-        deals with fuzziness in old-style (action/local_action) module invocations
+        deals with fuzziness in new style module invocations
+        accepting key=value pairs and dictionaries, and always returning dictionaries
         returns tuple of (module_name, dictionary_args)
```
Comment (mattclay) [inline lib/ansible/parsing/mod_args.py]: The return comment doesn't match the return value. Not a new problem, but should be fixed.

File: lib/ansible/parsing/mod_args.py [hunk 3]
```diff
@@ -178,15 +178,15 @@ def _normalize_parameters(self, thing, action=None, additional_args=dict()):
...
 
         possible example inputs:
-            { 'local_action' : 'shell echo hi' }
-            { 'action'       : 'shell echo hi' }
-            { 'local_action' : { 'module' : 'ec2', 'x' : 1, 'y': 2 }}
+            'echo hi', 'shell'
+            {'region': 'xyz'}, 'ec2'
         standardized outputs like:
             ( 'command', { _raw_params: 'echo hi', _uses_shell: True }
```
Comment (mattclay) [inline lib/ansible/parsing/mod_args.py]: This output comment is incorrect. Also not a new problem, but fix it while you're in the area.

**Review by mattclay — CHANGES_REQUESTED** [review]
I spotted two other corrections that should be made to the comments.

**Comment by willthames** [comment]
!needs_revision

**Comment by willthames** [comment]
ready_for_review

**Comment by willthames** [comment]
@mattclay I fixed your review points and also went a bit further

**Review by mattclay — APPROVED** [review]

**Comment by mattclay** [comment]
@willthames Merged! Thanks for cleaning this up.

**Pull request merged**